### PR TITLE
Speed up calls to get_local_windpower

### PR DESCRIPTION
**Pull request by OrenAudeles**
#### Summary
```SUMMARY: Performance "Speed up calls to get_local_windpower"```

#### Purpose of change
`get_local_windpower` is called a lot while processing fields through the lambda function `spread_gas`. For as many times as it was being called it wasn't super fast, and was taking a not insignificant amount of time to get results from.

#### Describe the solution
Firstly, test to see if we can exit early and skip any and all later expensive operations. If the tile is considered `sheltered` we can immediately return with a value of `0.0`

Secondly, for the precision required in `convert_wind_to_coord` we really didn't need to be using sin and cos functions which are known to be slow to calculate. The outputs are regular and predictable, made a table to iterate over, considerably faster. The table is listed in descending order of angle for ease of iteration.

#### Future Work
Can probably optimize the `oter_id`-based windpower multiplier application, and there should be a way to speed up the call to `is_wind_blocker`.

### CODE REVIEW

File: src/weather.cpp [hunk 1]
```diff
@@ -921,29 +923,24 @@ std::string get_wind_desc( double windpower )
...
+    const std::array<std::pair<int, rl_vec2d>, 9> outputs = {{
+            { 330, rl_vec2d( 0, -1 ) },
+            { 301, rl_vec2d( -1, -1 ) },
+            { 240, rl_vec2d( -1, 0 ) },
+            { 211, rl_vec2d( -1, 1 ) },
+            { 150, rl_vec2d( 0, 1 ) },
+            { 121, rl_vec2d( 1, 1 ) },
+            { 60, rl_vec2d( 1, 0 ) },
+            { 31, rl_vec2d( 1, 1 ) },
+            { 0, rl_vec2d( 0, -1 ) }
```
Comment (AMurkin) [inline src/weather.cpp]: Numbers are wrong
image

File: src/weather.cpp [hunk 2]
```diff
@@ -921,29 +923,24 @@ std::string get_wind_desc( double windpower )
...
-    } else {
-        roundedx = 0;
-    }
-    if( fy > 0.5 ) {
-        roundedy = 1;
-    } else if( fy < -0.5 ) {
-        roundedy = -1;
-    } else {
-        roundedy = 0;
+    const std::array<std::pair<int, rl_vec2d>, 9> outputs = {{
```
Comment (BevapDin) [inline src/weather.cpp]:
```suggestion
    static const std::array<std::pair<int, rl_vec2d>, 9> outputs = {{
```

File: src/weather.cpp (hunk 1 above)
Comment (OrenAudeles) [reply src/weather.cpp]: Numbers are accurate to previous implementation. I'm aware of the heavy bias towards cardinal directions with the 60 deg cardinal arcs and 30 deg diagonal arcs.

Comment (OrenAudeles) [reply src/weather.cpp]: I could reduce the bias pretty easily, but doing so is outside the scope of this PR.
Cardinal arcs would become 46 deg, with 23 deg half arcs
Diagonal arcs would become 44 deg, with 22 deg half arcs
These values are close enough to correct, and don't require changing from integer to floating point representation.

Comment (AMurkin) [reply src/weather.cpp]: Okay. Than only one thing: it should be `{ 31, rl_vec2d( 1, -1 ) }`.

Comment (OrenAudeles) [reply src/weather.cpp]: ah, thanks for catching that!

**Pull request merged**